core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -197,7 +211,11 @@ public final T nameResolverFactory(NameResolver.Factory resolverFactory) {
     Preconditions.checkState(directServerAddress == null,
         "directServerAddress is set (%s), which forbids the use of NameResolverFactory",
         directServerAddress);
-    this.nameResolverFactory = resolverFactory;
+    if (resolverFactory != null) {
```

Comment:
I strongly avoid unnecessary behavior changes in refactoring PRs, and hope others do the same. If you'd like to send out a PR or make an issue to change the behavior of `null`, that can be done.